cnf-tests/testsuites/e2esuite/multinetworkpolicy/multinetworkpolicy_sriov.go

```diff
@@ -0,0 +1,750 @@
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/sriov"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	SRIOV_RESOURCE    = "sriovnicMultiNetworkpolicyResource"
+	TEST_NETWORK_NAME = "test-multi-networkpolicy-sriov-network"
+	DEFAULT_NAMESPACE = "default"
+	TEST_NETWORK      = DEFAULT_NAMESPACE + "/" + TEST_NETWORK_NAME
+)
+
+// The following test model is common to all tests in the [multinetworkpolicy] test suite:
+// Three namespaces nsX, nsY, nsZ, each of them with three pods podA, podB, podC
+const nsX string = utils.MultiNetworkPolicyNamespaceX
+const nsY string = utils.MultiNetworkPolicyNamespaceY
+const nsZ string = utils.MultiNetworkPolicyNamespaceZ
+
+var nsX_podA, nsX_podB, nsX_podC *corev1.Pod
+var nsY_podA, nsY_podB, nsY_podC *corev1.Pod
+var nsZ_podA, nsZ_podB, nsZ_podC *corev1.Pod
+
+var port5555 intstr.IntOrString = intstr.FromInt(5555)
+var port6666 intstr.IntOrString = intstr.FromInt(6666)
+var protoTCP corev1.Protocol = corev1.ProtocolTCP
+var protoUDP corev1.Protocol = corev1.ProtocolUDP
+
+var _ = Describe("[sriov] MultiNetworkPolicy integration", func() {
+
+	sriovclient := sriovtestclient.New("")
+
+	execute.BeforeAll(func() {
+
+		if discovery.Enabled() {
+			Skip("Discovery is not supported.")
+		}
+
+		cleanPodsInNamespaces(sriovNamespaces.Test, nsX, nsY, nsZ)
+
+		Expect(sriovClean.All()).ToNot(HaveOccurred())
+		sriov.WaitStable(sriovclient)
+
+		createNamespaces(sriovNamespaces.Test, nsX, nsY, nsZ)
+
+		sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sriovInfos).ToNot(BeNil())
+
+		nodesList, err := nodes.MatchingOptionalSelectorByName(sriovInfos.Nodes)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(nodesList)).To(BeNumerically(">", 0))
+
+		sriovDevice, err := sriovInfos.FindOneSriovDevice(nodesList[0])
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = sriovNetwork.CreateSriovPolicy(sriovclient, "test-policy-", namespaces.SRIOVOperator,
+			sriovDevice.Name, nodesList[0], 10, SRIOV_RESOURCE, "netdevice")
+		Expect(err).ToNot(HaveOccurred())
+
+		sriov.WaitStable(sriovclient)
+
+		ipam := `{
+		         		"type": "host-local",
+		         		"subnet": "2.2.2.0/24",
+		         		"rangeStart": "2.2.2.8",
+		         		"rangeEnd": "2.2.2.67",
+		         		"routes": [{
+		           		"dst": "0.0.0.0/0"
+		         		}]
+		       	}`
+
+		err = sriovNetwork.CreateSriovNetwork(sriovclient, sriovDevice, TEST_NETWORK_NAME, DEFAULT_NAMESPACE,
+			namespaces.SRIOVOperator, SRIOV_RESOURCE, ipam)
+		Expect(err).ToNot(HaveOccurred())
+
+		networkAttachDef := netattdefv1.NetworkAttachmentDefinition{}
+		waitForObject(
```

Comment:
I think we have a function for this already in pkg if not I think we should move it there.

you also add the networkAttachDef scheme into the global client so you can use it from there no need to use it from the sriovClient

Reply:
Can't find anything similar.

Also, not sure why but it works also with global client without adding the scheme to it. It probably fallbacks to a not structured client.